src/app/utility/energy-consumption/energy-consumption.component.ts

```diff
@@ -61,31 +63,47 @@ export class EnergyConsumptionComponent implements OnInit {
     this.utilityDataSub.unsubscribe();
   }
 
-  checkMeterData(utilityMeterData) {
+  checkMeterData(utilityMeterData: Array<IdbUtilityMeterData>) {
+    this.hasElectricityError = false;
+    this.hasNaturalGasError = false;
+    this.hasOtherFuelsError = false;
+    this.hasOtherEnergyError = false;
+    this.hasOtherUtilityError = false;
+    this.hasWasteWaterError = false;
+    this.hasWaterError = false;
+
     this.utilityMeters.forEach(meter => {
-      const meterData = utilityMeterData.filter(meterData => { return meterData.meterId == meter.id });
-      if(meter.source === "Electricity") {
-        this.hasElectricityData = (meterData.length != 0 ? true : false);
+      if (meter.source === "Electricity" && !this.hasElectricityError) {
+        this.hasElectricityError = this.checkHasErrors(meter, utilityMeterData)
+      } 
+      if (meter.source === "Natural Gas" && !this.hasNaturalGasError) {
+        this.hasNaturalGasError = this.checkHasErrors(meter, utilityMeterData)
       }
-      if(meter.source === "Natural Gas") {
-        this.hasNaturalGasData = (meterData.length != 0 ? true : false);
+      if (meter.source === "Other Fuels" && !this.hasOtherFuelsError) {
+        this.hasOtherFuelsError = this.checkHasErrors(meter, utilityMeterData)
       }
-      if(meter.source === "Other Fuels") {
-        this.hasOtherFuelsData = (meterData.length != 0 ? true : false);
+      if (meter.source === "Other Energy" && !this.hasOtherEnergyError) {
+        this.hasOtherEnergyError = this.checkHasErrors(meter, utilityMeterData)
       }
-      if(meter.source === "Other Energy") {
-        this.hasOtherEnergyData = (meterData.length != 0 ? true : false);
+      if (meter.source === "Water" && !this.hasWaterError) {
+        this.hasWaterError = this.checkHasErrors(meter, utilityMeterData)
       }
-      if(meter.source === "Water") {
-        this.hasWaterData = (meterData.length != 0 ? true : false);
+      if (meter.source === "Waste Water" && !this.hasWasteWaterError) {
+        this.hasWasteWaterError = this.checkHasErrors(meter, utilityMeterData)
       }
-      if(meter.source === "Waste Water") {
-        this.hasWasteWaterData = (meterData.length != 0 ? true : false);
-      }
-      if(meter.source === "Other Utility") {
-        this.hasOtherUtilityData = (meterData.length != 0 ? true : false);
+      if (meter.source === "Other Utility" && !this.hasOtherUtilityError) {
+        this.hasOtherUtilityError = this.checkHasErrors(meter, utilityMeterData)
       }
     });
   }
 
+  checkHasErrors(meter: IdbUtilityMeter, facilityMeterData: Array<IdbUtilityMeterData>) {
+    let meterData: Array<IdbUtilityMeterData> = facilityMeterData.filter(data => { return data.meterId == meter.id });
+    let dataHasErrors: Date = this.utilityMeterDataService.checkForErrors(meterData);
+    if(dataHasErrors){
+      return true;
+    }
+    return meterData.length == 0;
```

Comment:
Why return meterData.length == 0? Shouldn't it be false if no errors?

Reply:
It was setup to display an error if a meter had no data. That's what this check is doing

Errors (for now):
1. duplicate month entries
2. meter has no data